src/super_gradients/modules/backbone_replacement_utils.py

```diff
@@ -0,0 +1,56 @@
+from typing import Union, Optional, Callable
+
+import torch
+from torch import nn
+
+__all__ = ["replace_input_channels_with_random_weights"]
+
+
+def compute_new_weights(
+    module: Union[nn.Conv2d, nn.Linear, nn.Module],
+    in_channels: int,
+    fn: Optional[Callable[[nn.Module, int], nn.Module]] = None,
+) -> nn.Module:
+    fn = fn or replace_input_channels_with_random_weights
+    return fn(module=module, in_channels=in_channels)
+
+
+def replace_input_channels_with_random_weights(module: Union[nn.Conv2d, nn.Linear, nn.Module], in_channels: int) -> nn.Module:
```

Comment:
So the typing supports nn.Linear but throws an error if it isnt conv?
It makes more sense to simply support a dedicated conv2d replacement by changing the name of this method, and dropping the ifs and exceptions.

Reply:
> So the typing supports nn.Linear but throws an error if it isnt conv?
That was a mistake, `nn.Linear` shouldnt be in typing

> It makes more sense to simply support a dedicated conv2d replacement by changing the name of this method, and dropping the ifs and exceptions.
Actually that's a good point, i'll change the name to be specifically on conv2d

Reply:
Changed here [622c61e](https://github.com/Deci-AI/super-gradients/pull/1557/commits/622c61ebaabf752d3f0081a1d738a5924d3f67f1)